docs(*): update-ai-assited-dev-npm-redme (#160)
IgniteUI/igniteui-react: .azure-pipelines/artifacts/README_Components.md

 
| Component | What it provides |
|:----------|:----------------|
| **Agent Skills** | Structured specialized knowledge and workflow files: import paths, component patterns, decision flows, generate from image design |
| **CLI MCP server** (`igniteui-cli`) | Documentation queries, API reference, project scaffolding tools via MCP |
| **Theming MCP server** (`igniteui-theming`) | Design tokens, palette tools, CSS custom property generation, WCAG AA contrast validation |
| **MAKER MCP server** (`@igniteui/maker-mcp`) | Multi-agent orchestration: decomposes complex tasks into validated, executable step plans |
 
> **Note:** All AI toolchain commands require Ignite UI CLI 15.0.0 or newer.
 
### Add AI toolchain to an existing project
 
Run this command from the root of your project:
 
```bash
ig ai-config
 
This adds Agent Skills and instructions to agent discovery path(s) and configures MCP server entries. 
 
### Start a new project with AI toolchain
 
Run the Ignite UI CLI to scaffold a new React project with the AI toolchain configured from the start:
 
```bash
ig new
```
> If the Ignite UI CLI is not installed globally, run `npm install -g igniteui-cli`.
 
### Agent Skills
 

> This package ships with **Agent Skills** — instructions, developer-owned structured knowledge files that tell AI coding assistants exactly how to use Ignite UI for React. The skill files are included in the [`igniteui-react`](https://www.npmjs.com/package/igniteui-react) package and also live in the [`skills/`](https://github.com/IgniteUI/igniteui-react/tree/master/skills) directory.

Reviewer: ```suggestion
This package ships with **Agent Skills** — structured knowledge files that tell AI coding assistants exactly how to use Ignite UI for React. The skill files are included in the [`igniteui-react`](https://www.npmjs.com/package/igniteui-react) package and also live in the [`skills/`](https://github.com/IgniteUI/igniteui-react/tree/master/skills) directory.
```

| Tree | ✅ | [Docs](https://www.infragistics.com/products/ignite-ui-react/react/components/grids/tree) |

## AI-Assisted Development
 
Ignite UI for React provides a toolchain to enhance the AI coding workflow — **Agent Skills**, the **Ignite UI CLI MCP server**, the **Ignite UI Theming MCP server**, and the **MAKER MCP server** — that grounds AI coding assistants in correct component APIs, import paths, and design tokens.
 
| Component | What it provides |
|:----------|:----------------|

> | **Agent Skills** | Developer-owned instruction packages: import paths, component patterns, decision flows, project conventions |

Reviewer: ```suggestion
| **Agent Skills** | Structured specialized knowledge and workflow files: import paths, component patterns, decision flows, generate from image design |
```

| **CLI MCP server** (`igniteui-cli`) | Documentation queries, API reference, project scaffolding tools via MCP |
| **Theming MCP server** (`igniteui-theming`) | Design tokens, palette tools, CSS custom property generation, WCAG AA contrast validation |
| **MAKER MCP server** (`@igniteui/maker-mcp`) | Multi-agent orchestration: decomposes complex tasks into validated, executable step plans |
 
> **Note:** All AI toolchain commands require Ignite UI CLI 15.0.0 or newer.
 
### Add AI toolchain to an existing project
 
Run this command from the root of your project:
 
```bash
ig ai-config

> ```

Reviewer: ````suggestion
```
> If the Ignite UI CLI is not installed globally, run `npm install -g igniteui-cli`.
````

> This copies Agent Skills into the agent discovery path and writes MCP server entries to `.vscode/mcp.json`. 

Reviewer: ```suggestion
This adds Agent Skills and instructions to agent discovery path(s) and configures MCP server entries. 
```

 
### Start a new project with AI toolchain
 
Run the Ignite UI CLI to scaffold a new React project with the AI toolchain configured from the start:
 
```bash
ig new
```

>  

Reviewer: ```suggestion
> If the Ignite UI CLI is not installed globally, run `npm install -g igniteui-cli`.
```

## AI-Assisted Development
 
Ignite UI for React provides a toolchain to enhance the AI coding workflow — **Agent Skills**, the **Ignite UI CLI MCP server**, the **Ignite UI Theming MCP server**, and the **MAKER MCP server** — that grounds AI coding assistants in correct component APIs, import paths, and design tokens.
 
| Component | What it provides |
|:----------|:----------------|
| **Agent Skills** | Structured specialized knowledge and workflow files: import paths, component patterns, decision flows, generate from image design |
| **CLI MCP server** (`igniteui-cli`) | Documentation queries, API reference, project scaffolding tools via MCP |
| **Theming MCP server** (`igniteui-theming`) | Design tokens, palette tools, CSS custom property generation, WCAG AA contrast validation |
| **MAKER MCP server** (`@igniteui/maker-mcp`) | Multi-agent orchestration: decomposes complex tasks into validated, executable step plans |
 
> **Note:** All AI toolchain commands require Ignite UI CLI 15.0.0 or newer.
 
### Add AI toolchain to an existing project
 
Run this command from the root of your project:
 
```bash
ig ai-config

> ```

Reviewer: ````suggestion
```

> If the Ignite UI CLI is not installed globally, run `npm install -g igniteui-cli`.
````

 
| Component | What it provides |
|:----------|:----------------|
| **Agent Skills** | Developer-owned instruction packages: import paths, component patterns, decision flows, project conventions |
| **CLI MCP server** (`igniteui-cli`) | Documentation queries, API reference, project scaffolding tools via MCP |
| **Theming MCP server** (`igniteui-theming`) | Design tokens, palette tools, CSS custom property generation, WCAG AA contrast validation |
| **MAKER MCP server** (`@igniteui/maker-mcp`) | Multi-agent orchestration: decomposes complex tasks into validated, executable step plans |
 
> **Note:** All AI toolchain commands require Ignite UI CLI 15.0.0 or newer.
 
### Add AI toolchain to an existing project
 
Run this command from the root of your project:
 
```bash
ig ai-config
```
 
This copies Agent Skills into the agent discovery path and writes MCP server entries to `.vscode/mcp.json`. 
 
### Start a new project with AI toolchain
 
Run the Ignite UI CLI to scaffold a new React project with the AI toolchain configured from the start:
 
```bash
ig new
```
 
 
### Agent Skills
 

> This package ships with **Agent Skills** — structured, developer-owned instruction packages that tell AI coding assistants exactly how to use Ignite UI for React. The skill files are included in the [`igniteui-react`](https://www.npmjs.com/package/igniteui-react) package and also live in the [`skills/`](https://github.com/IgniteUI/igniteui-react/tree/master/skills) directory.

Author: ```suggestion
This package ships with **Agent Skills** — instructions, developer-owned structured knowledge files that tell AI coding assistants exactly how to use Ignite UI for React. The skill files are included in the [`igniteui-react`](https://www.npmjs.com/package/igniteui-react) package and also live in the [`skills/`](https://github.com/IgniteUI/igniteui-react/tree/master/skills) directory.
```

## AI-Assisted Development
 
Ignite UI for React provides a complete AI toolchain — **Agent Skills**, the **Ignite UI CLI MCP server**, the **Ignite UI Theming MCP server**, and the **MAKER MCP server** — that grounds AI coding assistants in correct component APIs, import paths, and design tokens.
 
| Component | What it provides | Requires Ignite UI installed? |
|:----------|:----------------|:-----------------------------:|
| **Agent Skills** | Developer-owned instruction packages: import paths, component patterns, decision flows, project conventions | Yes |
| **CLI MCP server** (`igniteui-cli`) | Documentation queries, API reference, project scaffolding tools via MCP | No |
| **Theming MCP server** (`igniteui-theming`) | Design tokens, palette tools, CSS custom property generation, WCAG AA contrast validation | No |
| **MAKER MCP server** (`@igniteui/maker-mcp`) | Multi-agent orchestration: decomposes complex tasks into validated, executable step plans | No |
 
> **Note:** All AI toolchain commands require Ignite UI CLI 16.0.0 or newer.
 
### Add AI toolchain to an existing project
 
Run this command from the root of your project:
 
```bash
ig ai-config
```
 

> This copies Agent Skills into the agent discovery path and writes MCP server entries to `.vscode/mcp.json`. After the command finishes, start the MCP servers in your AI client — they are configured but not yet running.

Author: ```suggestion
This copies Agent Skills into the agent discovery path and writes MCP server entries to `.vscode/mcp.json`. 
```

 
### Start a new project with AI toolchain
 
Run the Ignite UI CLI to scaffold a new React project with the AI toolchain configured from the start:
 
```bash
ig new
```
 

> The CLI prompts for project name and theme, installs dependencies, and configures Agent Skills and MCP servers in a single interactive flow. If the CLI is not installed globally, use `npm install -g igniteui-cli`.

Author: ```suggestion
```

| Tree | ✅ | [Docs](https://www.infragistics.com/products/ignite-ui-react/react/components/grids/tree) |

## AI-Assisted Development
 

> Ignite UI for React provides a complete AI toolchain — **Agent Skills**, the **Ignite UI CLI MCP server**, the **Ignite UI Theming MCP server**, and the **MAKER MCP server** — that grounds AI coding assistants in correct component APIs, import paths, and design tokens.

Reviewer: Wouldn't go as far as say it's a complete or even a toolchain really.

You can say the Agent skills and MCP-s "enhance the AI coding [workflow]" for example.

Author: Ignite UI for React provides a toolchain to enhance the AI coding workflow -

Author: ```suggestion
Ignite UI for React provides a toolchain to enhance the AI coding workflow — **Agent Skills**, the **Ignite UI CLI MCP server**, the **Ignite UI Theming MCP server**, and the **MAKER MCP server** — that grounds AI coding assistants in correct component APIs, import paths, and design tokens.
```

 
| Component | What it provides | Requires Ignite UI installed? |
|:----------|:----------------|:-----------------------------:|

> | **Agent Skills** | Developer-owned instruction packages: import paths, component patterns, decision flows, project conventions | Yes |

Reviewer: Skills can be be delivered via `ig ai-config` without the package, that was the point and we're enhancing that to work even without existing project structure so the last value is also "no" here and TBH just drop the last column entirely - doesn't add useful info.

Reviewer: Also is project conversion supposed to be the image to design skill? Because it's a bit vague what that should be

| **CLI MCP server** (`igniteui-cli`) | Documentation queries, API reference, project scaffolding tools via MCP | No |
| **Theming MCP server** (`igniteui-theming`) | Design tokens, palette tools, CSS custom property generation, WCAG AA contrast validation | No |

> | **MAKER MCP server** (`@igniteui/maker-mcp`) | Multi-agent orchestration: decomposes complex tasks into validated, executable step plans | No |

Author: ```suggestion
| Component | What it provides |
|:----------|:----------------|
| **Agent Skills** | Developer-owned instruction packages: import paths, component patterns, decision flows, project conventions |
| **CLI MCP server** (`igniteui-cli`) | Documentation queries, API reference, project scaffolding tools via MCP |
| **Theming MCP server** (`igniteui-theming`) | Design tokens, palette tools, CSS custom property generation, WCAG AA contrast validation |
| **MAKER MCP server** (`@igniteui/maker-mcp`) | Multi-agent orchestration: decomposes complex tasks into validated, executable step plans |
```

> > **Note:** All AI toolchain commands require Ignite UI CLI 16.0.0 or newer.

Reviewer: You mean 15+, right? :)

> > **Note:** All AI toolchain commands require Ignite UI CLI 16.0.0 or newer.

Author: ```suggestion
> **Note:** All AI toolchain commands require Ignite UI CLI 15.0.0 or newer.
```

 
You can start evaluating Ignite UI AI assistance with the MCP servers alone — Ignite UI does not need to be installed in your project. Agent Skills become available once you install Ignite UI packages.

>  

Reviewer: ```suggestion
```

### Add AI toolchain to an existing project
 
Run this command from the root of your project:
 
```bash

> npx igniteui-cli ai-config

Reviewer: If you're giving instructions to install the package at the end and the other examples uses `ig`, use that consistently:

```suggestion
ig ai-config
```

```
 

> This copies Agent Skills into the agent discovery path and writes MCP server entries to `.vscode/mcp.json`. After the command finishes, start the MCP servers in your AI client — they are configured but not yet running.

Reviewer: Please refrain from such specifics. The command sets up configurations for AI coding assistants - namely it adds Ignite UI for React Agent Skills to the project and MCP servers setup (future release: and Agents instructions too).
The exact details of _how_ we add the skills, where we add them and which MCP servers are setup where are subject to change.

 
### Start a new project with AI toolchain
 
Run the Ignite UI CLI to scaffold a new React project with the AI toolchain configured from the start:
 
```bash
ig new
```
 

> The CLI prompts for project name and theme, installs dependencies, and configures Agent Skills and MCP servers in a single interactive flow. If the CLI is not installed globally, use `npm install -g igniteui-cli`.

Reviewer: Again don't see how explaining the name prompts and the install of deps is useful information. Just explain you can scaffold a new project with pre-configured AI setup, run `ig new` and follow the prompts.

The install of the package could be pulled ether before the two sections or added as a note under both snippets

 
### Agent Skills
 

> This package ships with **Agent Skills** — structured, developer-owned instruction packages that tell AI coding assistants exactly how to use Ignite UI for React. The skill files are included in the [`igniteui-react`](https://www.npmjs.com/package/igniteui-react) package and also live in the [`skills/`](https://github.com/IgniteUI/igniteui-react/tree/master/skills) directory.

Reviewer: I liked the original description paragraph we use across repos better, the "developer-owned instruction packages" says a lot of words, but can't tell what it means TBH.

Author: subjective

 
| Skill | Description |
|:------|:------------|
| **components** | Identify the right React components (`Igr*`) for a UI pattern, then install, import, and use them — JSX patterns, events, refs, forms, Next.js setup |
| **customize-theme** | Customize styling using CSS custom properties, Sass, and the theming system in React |
| **optimize-bundle-size** | Reduce bundle size with granular imports, tree-shaking, and lazy loading |
 

> When a Skill is active in the AI client, the agent follows the Skill instead of relying on general training data — which may reference outdated API signatures or import paths. The Skill package is developer-owned: edit the `SKILL.md` to match your team's conventions, add project-specific patterns, and version the package alongside your codebase.

Reviewer: Not sure we need to explicitly explain what that is, could easily just link one of the agent skills mentions to https://agentskills.io/home

Oh, and while true, not sure we want to encourage that since it interferes with update options and "version the package" is wrong since it's not a package, so delete the last part at least, if not the whole section

```suggestion
When a Skill is active in the AI client, the agent follows the Skill instead of relying on general training data — which may reference outdated API signatures or import paths.
```

 
### Supported AI Clients
 
| Client | Configuration method |
|:-------|:--------------------|
| VS Code with GitHub Copilot | `.vscode/mcp.json` |
| Cursor | `.cursor/mcp.json` |
| Claude Desktop (macOS) | `~/Library/Application Support/Claude/claude_desktop_config.json` |
| Claude Desktop (Windows) | `%APPDATA%\Claude\claude_desktop_config.json` |
| Claude Code | `.mcp.json` or the Claude Code MCP CLI command |
| JetBrains AI Assistant | **Tools → AI Assistant → Model Context Protocol (MCP)** |
 

> Agent Skills are compatible with GitHub Copilot via `.github/copilot-instructions.md`, Cursor via `.cursorrules` or `.cursor/rules/`, Windsurf via `.windsurfrules`, and JetBrains AI Assistant via project-level prompt settings.

Reviewer: Please no.

- We haven't release the update to pick different MCP configs yet, not sure where that list came from, besides being inaccurate.
- Skills are **not** _instructions_ (which we set up as well) or _rules_
- There are so many agents/tools and a lot of them support skills and almost all of them work with `.agents/skills` (except claude, which we cover as well), so you can't put a definitive list.
```suggestion
```
